server/src/internalClusterTest/java/org/elasticsearch/indices/recovery/IndexRecoveryIT.java

```diff
@@ -1614,6 +1627,127 @@ public void testReservesBytesDuringPeerRecoveryPhaseOne() throws Exception {
         );
     }
 
+    public void testWaitForClusterState() throws Exception {
```

Comment:
nit: maybe we should rename this test to `testWaitForClusterStateToBeAppliedInSourceNode`?

Reply:
++ done in 55ace3e1aafb840cbe2647884f04b1e6055221d3